x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/agents/agent_list_page/components/migrate_agent_flyout/index.tsx

```diff
@@ -122,17 +143,64 @@ export const AgentMigrateFlyout: React.FC<Props> = ({ agents, onClose, onSave })
             <h1>
               <FormattedMessage
                 id="xpack.fleet.agentList.migrateAgentFlyout.title"
-                defaultMessage="Migrate Agent"
+                defaultMessage="Migrate {plural}"
+                values={{
+                  plural: agents.length > 1 ? 'agents' : 'agent',
```

Comment:
you can directly use plural in `defaultMessage` like this :https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/agent_policy/components/confirm_deploy_modal.tsx#L98